rundetection/rules/common_rules.py

```diff
@@ -41,3 +49,83 @@ def verify(self, job_request: JobRequest) -> None:
                 "If it is a scatter, contains empty or direct in the title and is assumed to be a scatter "
                 "for an empty can run."
             )
+
+
+class MolSpecStitchRule(Rule[bool]):
+    """
+    Enables Tosca, Osiris, and Iris Run stitching
+    """
+
+    @staticmethod
+    def _is_title_similar(title: str, other_title: str) -> bool:
+        """
+        Compare one run title to another to check for similarity
+        :param title:the first run title
+        :param other_title:the second run title
+        :return: (bool) True if similar False otherwise
+        """
+        logger.info("Comparing titles %s and %s", title, other_title)
+        if title == other_title:
+            return True
+        if title[:-5] == other_title[:-5]:
+            return True
+        if title[0:7] == other_title[0:7] and ("run" in other_title or "run" in title):
+            return True
+        logger.info("Titles not similar, continuing")
+        return False
+
+    def _get_runs_to_stitch(self, run_path: Path, run_number: int, run_title: str, instrument: str) -> list[int]:
+        run_numbers = []
+        while run_path.exists():
+            logger.info("run path exists %s", run_path)
+            if not self._is_title_similar(get_run_title(run_path), run_title):
+                logger.info("titles not similar")
+                break
+            logger.info("titles are similar appending run number %s", run_number)
+            run_numbers.append(run_number)
+            run_number -= 1
+            if instrument.upper() == "TOSCA":
+                next_run_number = f"TSC{run_number}.nxs"
+            else:
+                next_run_number = f"{instrument}{run_number:08d}.nxs"
```

Comment:
Although I imagine the instrument value is always in upper case but just to be anal...

How come you're checking that the instrument is upper when it's Tosca but with Osiris it's treated as a given that it's already upper case?

Reply:
Tosca doesn't use the instrument name but TSC, we should probably add .upper() to it so I shall do that.
```suggestion
            if instrument.upper() == "TOSCA":
                next_run_number = f"TSC{run_number}.nxs"
            else:
                next_run_number = f"{instrument.upper()}{run_number:08d}.nxs"
```